### PR TITLE
Task5: Create endpoint getAllArticles

### PR DESCRIPTION
**Pull request by ccdu1010**
-- add api endpoint for getting all articles
-- group comment_count with the article id which in descending order
-- error handling
-- created api tests

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -99,9 +99,44 @@ describe("app", () => {
...
+    return request(app).get(`/api/articles`).expect(200);
+  });
+  test("200: responds with correct comment_count on related actricle_id", () => {
+    return request(app)
+    .get(`/api/articles`)
+    .expect(200)
+    .then((response) => {
+      const articleIdToTest = 34;
+      const article = response.body.articles.filter(article => article.article_id === articleIdToTest);
+      const expectedCommentCount = 11;
```
Comment (Augs0) [inline __tests__/app.test.js]: I'm not sure if this test block is unfinished or if you were trying to do something global but  it isn't really doing anything at the moment as there is no assertion

File: __tests__/app.test.js [hunk 2]
```diff
@@ -99,9 +99,44 @@ describe("app", () => {
...
+    });
+  });
+  test("200: respond with an article object with the following properties when article objects are found", () => {
+    return request(app)
+    .get(`/api/articles`)
+    .expect(200)
+    .then((response) => {
+      const {articles} = response.body;
+      articles.forEach((article) => {
```
Comment (Augs0) [inline __tests__/app.test.js:133]: Whenever using iteration, we should check the length of the array first

File: __tests__/app.test.js [hunk 3]
```diff
@@ -99,9 +99,44 @@ describe("app", () => {
...
+      expect(article).toHaveProperty("article_id", expect.any(Number));
+      expect(article).toHaveProperty("topic", expect.any(String));
+      expect(article).toHaveProperty("created_at", expect.any(String));
+      expect(article).toHaveProperty("votes", expect.any(Number));
+      expect(article).toHaveProperty("article_img_url", expect.any(String));
+      expect(article).toHaveProperty("comment_count", expect.any(String));
+      expect(article).not.toHaveProperty("body");
+    });
+  });
+ });
```
Comment (Augs0) [inline __tests__/app.test.js:146]: Good start. We also need to test if these articles are returned, by default, in descending order by created_at

File: controllers/articles.controller.js [hunk 4]
```diff
@@ -16,4 +17,18 @@ const getArticleById = (request, response, next) => {
   .catch(next);
 };
 
-module.exports = { getArticleById }
\ No newline at end of file
+exports.getAllArticles = (request, response, next) => {
+  selectAllArticles()
+  .then((articles) => {
+    if(!articles || articles.length === 0) {
+        return Promise.reject({
```
Comment (Augs0) [inline controllers/articles.controller.js]: Please handle this 404 in the model

**Review by Augs0 — CHANGES_REQUESTED** [review]
A really good effort! Thee are a few things I'd like you to go back and change or add. Please resubmit once you have done the things mentioned in my comments

**Review by leekli — APPROVED** [review]
Thanks for making these changes CC.

You are can now merge this branch and begin working on your katas.

**Pull request merged**